### PR TITLE
Fixes Vampire things

### LINKED ISSUE #19816: Datums Vampire scoreboards don't space out/add a return line between the vampire and its thralls.
See picture
capture

### LINKED ISSUE #19817: Datums vampire faction icons no worky
With such amazing feats like thralls not being able to tell who their masters/fellow thralls are.

### LINKED ISSUE #19819: Datums Vampire have 0 starting blood, which cucks them out of Glare & Hypnotise
Apparently you need 1 blood to cast Glare, but you ain't got no blood, so no casting that. I hope you like I-CAN'T-BELIEVE-IT'S-NOT-CHLORAL (blocked by chest armor) or tase-n-succ to be able to cast glare.

### LINKED ISSUE #19844: Datum vampires lose their active abilities after a brain transplant
As stated on the topic. You're still a vampire with all holy and sunlight weaknesses plus the ability to suck blood but all active abilities vanish. No hud icons or ability tab, reconnecting did not fix this.

### LINKED ISSUE #19880: Vampires losetheir powers upon being cloned.
See title, pulled from the thread 
capture

### PR DESCRIPTION
**Pull request by ShiftyRail**
Closes #19819
Closes #19817 
Closes #19816 
Closes #19844 
Closes #19880 

Tested and ready to be merged

:cl:
- bugfix: Fixed problems with vampires (starting blood, faction icons, and bats attacking their master).

### CODE REVIEW

File: code/datums/gamemode/role/role.dm [hunk 1]
```diff
@@ -105,6 +105,8 @@
 	faction=fac
 	if(!faction)
 		ticker.mode.orphaned_roles += src
+	else
+		faction.members += src
```
Comment (MadmanMartian) [inline code/datums/gamemode/role/role.dm:109]: Highly illegal

**Review by MadmanMartian — CHANGES_REQUESTED** [review]
Regarding the thrall-vampire thing:

Why not have the protect master be a faction objective, so you don't have to hold a reference to the master in the thrall datum

File: code/datums/gamemode/role/role.dm (hunk 1 above)
Comment (ShiftyRail) [reply code/datums/gamemode/role/role.dm:109]: ?
If they are linked to a faction why not add them to the members

**Comment by ShiftyRail** [comment]
image

**Comment by ShiftyRail** [comment]
@MadmanMartian I've fixed the vampire stone mask which was adding people to its faction manually
Everyone else uses `HandleRecreuitedMind` which is functionally strictly equivalent

**Comment by ShiftyRail** [comment]
image

Fixes #19844, but for cloning only, so don't merge this yet.
I've found a way to fix it on brain transplant (ie, on mind transfer).

**Comment by DamianX** [comment]
Should probably call a more generic role datum proc rather than specifically dealing with vampires in `transfer_to`

**Comment by ShiftyRail** [comment]
@DamianX You're right

**Comment by tayachiya** [comment]
Why is this not merged yet? If its tested and ready why do you keep vampires gimped and near unplayable?

**Comment by MadmanMartian** [comment]
It was merged on the 30th of september

**Pull request merged**